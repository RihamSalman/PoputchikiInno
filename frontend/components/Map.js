import React from "react";
import { StyleSheet, Text, View } from "react-native";
import MapView, { Marker } from "react-native-maps";
// import { selectOrigin } from "../slices/navSlice";
// import { useSelector } from "react-redux";
import tw from "tailwind-react-native-classnames";

const Map = () => {
  //   const origin = useSelector(selectOrigin);
  // console.log({ origin });
  // if (origin == null) {
  //   return null;
  // }
  return (
    <MapView
      style={tw`flex-1`}
      mapType="mutedStandard" // Simplify the map
      region={{
        latitude: 37.78825,
        longitude: -122.4324,
        latitudeDelta: 0.015,
        longitudeDelta: 0.0121,
      }}
      // initialRegion={{
        // Error: Null is not an object
        // latitude: origin.location.lat,
        // longitude: origin.location.lng,
        // latitudeDelta: 0.0922,
        // longitudeDelta: 0.0421,
      // }}
      
    />
  );
};

export default Map;

const styles = StyleSheet.create({});