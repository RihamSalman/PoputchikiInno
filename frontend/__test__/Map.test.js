import React from 'react';
import renderer from 'react-test-renderer';
import Map from '../components/Map'

test('renders correctly', () => {
    const tree = renderer.create(<Map />).toJSON();
    expect(tree).toMatchSnapshot();
  });