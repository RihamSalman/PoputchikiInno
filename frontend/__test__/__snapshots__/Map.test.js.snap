// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<AIRMap
  handlePanDrag={false}
  initialRegion={null}
  mapType="mutedStandard"
  onChange={[Function]}
  onLayout={[Function]}
  onMapReady={[Function]}
  onMarkerPress={[Function]}
  region={
    Object {
      "latitude": 37.78825,
      "latitudeDelta": 0.015,
      "longitude": -122.4324,
      "longitudeDelta": 0.0121,
    }
  }
  style={
    Object {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
/>
`;
